pycfmodel/model/resources/sns_topic_policy.py

```diff
@@ -12,19 +12,30 @@
 CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 """
+import json
+
+from typing import ClassVar, List
+
+from pydantic import validator
+
+from ..types import ResolvableStr
+from ..base import CustomModel
 from .resource import Resource
 from .properties.policy_document import PolicyDocument
 
 
-class SNSTopicPolicy(Resource):
-    def __init__(self, logical_id, value):
-        """
-        "PolicyDocument" : PolicyDocument,
-        "Topics" : [ List of SNS topic ARNs, ... ]
-        """
-        super().__init__(logical_id, value)
+class SNSTopicPolicyProperties(CustomModel):
+    PolicyDocument: PolicyDocument
+    Topics: List[ResolvableStr]
+
+    @validator("PolicyDocument", pre=True)
+    def handle_dual_policy_document(cls, value):
```

Comment:
Shouldn't this method apply to all other models too?

Reply:
Actually, this has to be removed. It was a test.